Tools/AP_Periph/can.cpp

```diff
@@ -727,643 +597,37 @@ void AP_Periph_FW::can_safety_button_update(void)
     pkt.button = ARDUPILOT_INDICATION_BUTTON_BUTTON_SAFETY;
     pkt.press_time = counter;
 
-    uint8_t buffer[ARDUPILOT_INDICATION_BUTTON_MAX_SIZE] {};
-    uint16_t total_size = ardupilot_indication_Button_encode(&pkt, buffer, !canfdout());
-
-    canard_broadcast(ARDUPILOT_INDICATION_BUTTON_SIGNATURE,
-                            ARDUPILOT_INDICATION_BUTTON_ID,
-                            CANARD_TRANSFER_PRIORITY_LOW,
-                            &buffer[0],
-                            total_size);
+    dronecan->button_pub.broadcast(pkt);
 }
 #endif // HAL_GPIO_PIN_SAFE_BUTTON
 
-/**
- * This callback is invoked by the library when a new message or request or response is received.
- */
-void AP_Periph_FW::onTransferReceived(CanardInstance* canard_instance,
-                                      CanardRxTransfer* transfer)
+void AP_Periph_DroneCAN::handle_restart_node(const CanardRxTransfer& transfer, const uavcan_protocol_RestartNodeRequest &req)
 {
-#ifdef HAL_GPIO_PIN_LED_CAN1
-    palToggleLine(HAL_GPIO_PIN_LED_CAN1);
```

Comment:
lost some of the LED support?

Reply:
moved to AP_Canard_Iface

Reply:
I don't see any references to HAL_GPIO_PIN_LED_CAN1 in there, but it is defined by kha_eth and by ZubaxGNSS. Did I miss an indirection somewhere?

Reply:
yeah, I forgot to commit it, its been added now.